CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Contributors' Guide
+
+This guide will help familiarize contributors to the `containerd/containerd` repository.
+
+## Prerequisite
+
+First read the containerd project's [general guidelines around contribution](https://github.com/containerd/project/blob/main/CONTRIBUTING.md)
+which apply to all containerd projects.
+
+## Getting started
+
+See [`BUILDING.md`](https://github.com/containerd/containerd/blob/main/BUILDING.md) to setup a local development environment.
+
+If you are also a new user to containerd, you can first check out the [_Getting starting with containerd_](https://github.com/containerd/containerd/blob/main/docs/getting-started.md) guide.
+
+## Setting up your environment
+
+At a minimum, the dev tools from `script/setup/install-dev-tools` should be installed.
+Run `make install-deps` to install dependencies used for running and developing the CRI plugin.
+Other install scripts under `script/setup` may need to be run depending on your environment and your preference for installing libraries and dependencies.
+The versions used by `containerd/containerd` CI can be found in `script/setup` and referred to if installing manually.
+
+```
+$ script/setup/install-dev-tools
+$ make install-deps
+```
+
+## Code style
+
+- Go files adhere to standard Go formatting and styling
+- Protobuf files use tabs for indentation
+- Other files must not contain trailing whitespace and should end with a single new line character
+
+Use the `check` command in the makefile to verify your code matches the expected style.
+
+```
+make check
+```
+
+## Updating protobuf files
+
+Ensure protoc and dev tools have been installed, then run `make protos`
+
+> **Note**
+> When running `make protos`, the current working directory should be found under the `GOPATH` environment
+> variable to ensure protoc can properly resolve the paths of protofiles in the project.
+
+## Naming packages
+
+Package names should be short and simple. Avoid using `_` and repeating words from parent directories.
+
+### Where to put packages
+
+Try to put a new package under the appropriate root directories. The root directory is reserved for
+configuration and build files, no source files will be accepted in root since containerd v2.0.
+
+- `api` - All protobuf service definitions and types used by services
```

Comment:
What will happen to `api` directory when v2 is released?

Reply:
I think the automatic syncing still makes sense and could apply for any non-go bindings. I'm still not sure how much we will want to rely on go submodules as well, not going to define that here and hopefully most contributors don't have to worry about it.